cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java

```diff
@@ -135,6 +156,10 @@ public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(deviceType.name());
         dest.writeInt(platformId);
         dest.writeInt(productId);
+        Parcelables.writeBoolean(dest, cellular);
+        dest.writeString(imei);
```

Comment:
Looks like I did that wrong on the `version` field, too.  Might be worth looking for all the nullable fields on these parcelables and applying `read/writeValue()` as needed.